core/src/impl/Kokkos_Profiling.hpp

```diff
@@ -142,6 +142,28 @@ void parseArgs(const std::string&);
 
 Kokkos_Profiling_SpaceHandle make_space_handle(const char* space_name);
 
+/**
+ * Convenience wrapper around kokkosp_mark_kernel_static_info
+ *
+ * Consider using markKernelStaticInfo<Functor>(kernelID) instead
+ */
+void markKernelStaticInfo(uint64_t kernelID, const KernelStaticInfo& info);
+
+/**
+ * Take a kernelID produced by e.g. beginParallelFor
+ * and associate compile-time information about Functor with it
+ *
+ * Arguments:
+ *
+ * kernelID: An ID for a parallel loop registered with e.g. beginParallelFor
+ */
+template <typename Functor>
+void markKernelStaticInfo(uint64_t kernelID) {
+  Kokkos::Tools::KernelStaticInfo info;
+  info.functor_size = sizeof(Functor);
+  markKernelStaticInfo(kernelID, info);
+}
+
```

Comment:
Do we really need both overloads? Can't we just inline the first one into the second one?

Reply:
ping

Reply:
first one does what a lot of the other profiling code does and calls `invoke_kokkosp_callback` which is a function template defined in `Kokkos_Profiling.cpp` and a declaration is not available in this header. I could move the entire implementation of that function into this header and do it the way you're suggesting if you prefer.